src/libcore/option.rs

```diff
@@ -1064,6 +1150,13 @@ fn expect_failed(msg: &str) -> ! {
     panic!("{}", msg)
 }
 
+// This is a separate function to reduce the code size of .expect_none() itself.
+#[inline(never)]
+#[cold]
+fn expect_none_failed<T: fmt::Debug>(msg: &str, value: T) -> ! {
```

Comment:
This could potentially take a `&dyn Debug` to cut down some potential code bloat?

Reply:
I suppose so -- as is, it's basically the same as `core::result::unwrap_failed`. We could even `pub(crate)` that and make them all go through `&dyn Debug`.

Reply:
OK, those functions in both `option` and `result` now take `&dyn Debug`.